types/node/index.d.ts

```diff
@@ -797,82 +809,93 @@ declare module "http" {
     /**
      * @deprecated Use IncomingMessage
      */
-    export interface ServerRequest extends IncomingMessage {
+    export class ServerRequest extends IncomingMessage {
         connection: net.Socket;
     }
 
-    export interface ServerResponseHeaders {
-      [key: string]: number | string | string[];
-    }
+    // https://github.com/nodejs/node/blob/master/lib/_http_outgoing.js
+    export class OutgoingMessage extends stream.Writable {
+        output: any[];
```

Comment:
I think we should remove `upgrading`. `chunkedEncoding`. `shouldKeepAlive`, and `useChunkedEncodingByDefault` as they are undocumented and implementation details.
And `socket` (currently in `ClientRequest`) should be here.